src/Jackalope/Transport/DoctrineDBAL/Client.php

```diff
@@ -1539,8 +1539,13 @@ private function getSystemIdForNode($identifier, $workspaceName = null)
         if (UUIDHelper::isUUID($identifier)) {
             $query = 'SELECT id FROM phpcr_nodes WHERE identifier = ? AND workspace_name = ?';
         } else {
-            if ($this->getConnection()->getDriver() instanceof \Doctrine\DBAL\Driver\PDOMySql\Driver) {
-                $query = 'SELECT id FROM phpcr_nodes WHERE path COLLATE utf8_bin = ? AND workspace_name = ?';
+            $params = $this->conn->getParams();
+            $charset = isset($params['charset']) ? $params['charset'] : 'utf8';
```

Comment:
oh well. but in that case lets expect people to set the case sensitive encoding manually?

i think we really must avoid doing a database call in every request to know this, so it must come from the connection / driver if thats possible at all. worst case we just add this new method and have a compiler pass in the phcpr-bundle that reads the dbal configuration and does something sensible?

@lsmith77 can you help us here to read the right information?